### DISCUSSION TITLE
Changing background for individual elements

### DISCUSSION THREAD

**Discussion by phbao27uet** [barvian/number-flow#42] [Help]
Hello, I'm still trying to use your library. However, I'm having some difficulties figuring out how to change the background of each element. I can use the image below as an example.

![Screenshot 2024-09-06 at 11 06 24](https://github.com/user-attachments/assets/8e1bbe65-51d2-4e6d-8481-4e25eb075d1a)

**Comment by barvian** [comment]
Hi, this is unfortunately not possible right now and it would be a pretty substantial effort to get it working smoothly with the animations 🫤

**Comment by rcoopr** [comment]
Maybe an aside to this, but I would be interested in applying styles to the integer/fractional parts separately. For example, having the numbers after a decimal point being a smaller font size for example.

As far as I understand, styles are siolated by virtue of the shadowRoot. So an option like `innerClasses` would never be useful, but perhaps we could pass some `innerStyles` vaguely like this?

```ts
const innerStyles = {
  integer: {
    'font-size': '1.2em'
  },
  fraction: {
    'font-size': '0.8em'
  }
}
```

I'd be happy to work on it but maybe there's a better API for that

**Reply by barvian** [reply]
Ah yeah, see #21 for more info on that 👍
